src/test/regress/expected/multi_colocation_utils.out

```diff
@@ -0,0 +1,310 @@
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1300000;
+ALTER SEQUENCE pg_catalog.pg_dist_jobid_seq RESTART 1300000;
+-- ===================================================================
+-- create test utility function
+-- ===================================================================
+CREATE SEQUENCE colocation_test_seq
+    MINVALUE 1
+    NO CYCLE;
+/* a very simple UDF that only sets the colocation ids the same 
+ * DO NOT USE THIS FUNCTION IN PRODUCTION. It manually sets colocationid column of
+ * pg_dist_partition and it does not check anything about pyshical state about shards.
+ */
+CREATE OR REPLACE FUNCTION colocation_test_colocate_tables(source_table regclass, target_table regclass)
+    RETURNS BOOL
+    LANGUAGE plpgsql
+    AS $colocate_tables$
+DECLARE nextid BIGINT;
+BEGIN
+    SELECT nextval('colocation_test_seq') INTO nextid;
+
+    UPDATE pg_dist_partition SET colocationId = nextid
+    WHERE logicalrelid IN
+    (
+        (SELECT p1.logicalrelid
+            FROM pg_dist_partition p1, pg_dist_partition p2
+            WHERE
+                p2.logicalrelid = source_table AND
+                (p1.logicalrelid = source_table OR 
+                (p1.colocationId = p2.colocationId AND p1.colocationId != 0)))
+        UNION
+        (SELECT target_table)
+    );
+    RETURN TRUE;
+END;
+$colocate_tables$;
+-- ===================================================================
+-- create test functions
+-- ===================================================================
+CREATE FUNCTION get_table_colocation_id(regclass)
```

Comment:
Could you add tests for FindShardIndex() as well? (since that is a critical function for us)